juju/juju
# Openstack - Include Network Cloud Config for Correct NICs

## Description of change

It was observed that although we correctly attached O7k ports to instances based on spaces constraints, the on-machine devices were not available.

This change ensures that Netplan is rendered correctly on multi-NIC OpenStack instances in order that NICs reflect the ports that were added to them.

We previously rendered a nil `CloudConfig` for O7k servers. Now we generate a new one and include network config that reflects the ports that we attach.

## QA steps

Ensure you have a suitable OpenStack for testing multi-NIC support:
https://discourse.jujucharms.com/t/openstack-multi-space-support/2672

After adding a machine with multi-space constraints, run:
```bash
juju run --machine 5 -- ip a
```

The interfaces should be in the up state and reflect the addresses/subnets that we confined the machine to.

## Documentation changes

None

## Bug reference

N/A
